packages/core/src/types.ts

```diff
@@ -207,3 +208,48 @@ export type ClientStorageType =
   | 'react-native'
   | 'web'
   | 'cookie'
+
+// Hasura-auth API response types
+interface NullableErrorResponse {
+  error: ErrorPayload | null
+}
+
+export type NhostSessionResponse =
```

Comment:
I'd change this type to this:

```ts
export interface NhostSessionResponse {
  /**
   * some documentation...
   */
  session: NhostSession | null
  /**
   * some documentation...
   */
  error: ErrorPayload | null
}
```

Reply:
we would loose the inference "if there is an error then the session is null"

Reply:
Yes, that's true. Then https://github.com/nhost/nhost/pull/907#discussion_r958083789 is also irrelevant.